content/courses/advanced-ada/parts/resource_management/anonymous_access_types.rst

```diff
@@ -252,21 +254,553 @@ cover more specific details on anonymous access types.
+
+If we hadn't initialized :ada:`AI` explicitly with :ada:`I'Access`, it would
+have been set to :ada:`null`, which would fail the :ada:`not null` constraint
+of the anonymous access type. Similarly, we also have to initialize the
+:ada:`Arr` array and return a valid access object for the
+:ada:`Init_Access_Integer` function.
 
 
+.. _Adv_Ada_Drawbacks_Anonymous_Access_To_Object_Types:
+
 Drawbacks of Anonymous Access-To-Object Types
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+Anonymous access-to-object types have important drawbacks. For example, some
+features that are available for named access types aren't available for the
+anonymous access types. Also, most of the drawbacks are related to how
+anonymous access-to-object types can potentially make the allocation and
+deallocation quite complicated or even error-prone.
+
+For starters, some pool-related features aren't available for anonymous
+access-to-object types. For example, we cannot specify which pool is going to
+be used in the allocation of an anonymous access-to-object. In fact, the memory
+pool selection is compiler-dependent, so we cannot rely on an object being
+allocated in a specific pool when using :ada:`new` with an anonymous
+access-to-object type. (In contrast, as we know, each named access type has an
+associated pool, so objects allocated via :ada:`new` will be allocated on that
+pool.) Also, we cannot identify which pool was selected for the allocation of a
+specific object, so we don't have any information to use for the deallocation of
+that object.
+
+Because the pool selection is hidden from us, this makes the memory
+deallocation more complicated. For example, we cannot instantiate the
+:ada:`Ada.Unchecked_Deallocation` procedure for anonymous access types. Also,
+some of the methods we could use to circumvent this limitation are error-prone,
+as we discuss in this section.
+
+Also, storage-related features are available: specifying the storage size |mdash|
+especially, specifying that the access type has a storage size of zero |mdash|
+isn't possible.
+
+Missing features
+^^^^^^^^^^^^^^^^
+
+Let's see a code example that shows some of the features that aren't available
+for anonymous access-to-object types:
+
+.. code:: ada compile_button project=Courses.Advanced_Ada.Resource_Management.Anonymous_Access_Types.Anonymous_Access_To_Object_Types.Missing_Anonymous_Access_To_Object_Features
+
+    with Ada.Unchecked_Deallocation;
+
+    package Missing_Features is
+
+       --  We cannot specify which pool will be used
+       --  in the anonymous access-to-object
+       --  allocation; the pool is selected by the
+       --  compiler:
+       IA : access Integer := new Integer;
+
+       --
+       --  All the features below aren't available
+       --  for an anonymous access-to-object:
+       --
+
+       --  Having a specific storage pool associated
+       --  with the access type:
+       type Integer_Access is
+          access Integer;
```

Comment:
Shouldn't the pool be specified, thus showing a feature not available for anonymous access types? I'm thinking of the following:

P : My_Pool;  -- a type derived from Root_Storage_Pool
type Integer_Access is access Integer;
for Integer_Access'Storage_Pool use P;

Reply:
I'm afraid that deriving from `Root_Storage_Pool` here would render the example too complicated. However, I like the idea of using the `Storage_Pool` attribute / aspect, so I'll add an example using the pool from another access type.